core/src/gl/error.h

```diff
@@ -18,10 +18,18 @@ class Error {
      */
     static bool hadGlError(const std::string& _locationTag);
 
+    static void glError(const char* stmt, const char* fname, int line);
+
 private:
 
     static std::unordered_map<GLenum, std::string> s_GlErrorCodesToStrings;
 
 };
 
+#ifdef TANGRAM_DEBUG
+#define GL_CHECK(STMT) do { STMT; Tangram::Error::glError(#STMT, __FILE__, __LINE__); } while (0)
```

Comment:
mapbox-gl-native has a pretty clever approach here to allow this macro to return the same value as the function it wraps: https://github.com/mapbox/mapbox-gl-native/blob/master/include/mbgl/gl/gl.hpp#L71

This may be *too* clever. I'm not sure how much the lambda calls would impact performance, but it does simplify usage of the macro. 

At a minimum, we should be able to simply write `GL_CHECK()` if we wrap `STMT` in another pair of parentheses, I think.